service/gateway/query_handler.go

```diff
@@ -196,3 +200,87 @@ func (g *Gateway) listBucketReadRecordHandler(w http.ResponseWriter, r *http.Req
 	}
 	log.Debugw("list bucket read records", "xml_info", xmlInfo)
 }
+
+// getObjectPutStateHandler handles the get put object state request
+func (g *Gateway) getObjectPutStateHandler(w http.ResponseWriter, r *http.Request) {
+	var (
+		err            error
+		errDescription *errorDescription
+		reqContext     *requestContext
+		addr           sdktypes.AccAddress
+		jobState       servicetypes.JobState
+		jobStateStr    string
+	)
+
+	reqContext = newRequestContext(r)
+	defer func() {
+		if errDescription != nil {
+			_ = errDescription.errorResponse(w, reqContext)
+		}
+		if errDescription != nil && errDescription.statusCode != http.StatusOK {
+			log.Errorf("action(%v) statusCode(%v) %v", getObjectPutStateRouterName, errDescription.statusCode, reqContext.generateRequestDetail())
+		} else {
+			log.Infof("action(%v) statusCode(200) %v", getObjectPutStateRouterName, reqContext.generateRequestDetail())
+		}
+	}()
+
+	if g.uploader == nil {
+		log.Error("failed to get bucket read quota due to not config uploader")
+		errDescription = NotExistComponentError
+		return
+	}
+
+	if err = s3util.CheckValidBucketName(reqContext.bucketName); err != nil {
+		log.Errorw("failed to check bucket name", "bucket_name", reqContext.bucketName, "error", err)
+		errDescription = InvalidBucketName
+		return
+	}
+	if err = s3util.CheckValidObjectName(reqContext.objectName); err != nil {
+		log.Errorw("failed to check object name", "object_name", reqContext.objectName, "error", err)
+		errDescription = InvalidKey
+		return
+	}
+
+	if addr, err = reqContext.verifySignature(); err != nil {
+		log.Errorw("failed to verify signature", "error", err)
+		errDescription = makeErrorDescription(err)
+		return
+	}
+	if err = g.checkAuthorization(reqContext, addr); err != nil {
+		log.Errorw("failed to check authorization", "error", err)
+		errDescription = makeErrorDescription(err)
+		return
+	}
+	if reqContext.objectInfo.GetObjectStatus() == storagetypes.OBJECT_STATUS_SEALED {
+		jobStateStr = storagetypes.OBJECT_STATUS_SEALED.String()
+	} else {
+		if jobState, err = g.uploader.QueryObjectPutState(context.Background(), reqContext.objectInfo.Id.Uint64()); err != nil {
+			log.Errorw("failed to query object put state", "error", err)
+			errDescription = makeErrorDescription(err)
+			return
+		}
+		jobStateStr = jobState.String()
```

Comment:
It's better to convert this internal JosState to other more readable state to users. Otherwise, users needs to understand the jobs inside SP.

Reply:
Yes, has added a description field.

Reply:
I see. 

It's still a bit complex for user to understand these Job State inside SP.  How about these progress status.

```
UPLOAD_PROGRESS_META_CREATED
UPLOAD_PROGRESS_DATA_UPLOADING
UPLOAD_PROGRESS_DATA_REPLICATING
UPLOAD_PROGRESS_SEALING
UPLOAD_PROGRESS_COMPLETE
UPLOAD_PROGRESS_FAILED
```

Reply:
Yes; a version has been updated, if you have time, please continue to review it~